scripts/core/ui/components/video-hls.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import Hls from 'hls.js'
+
+
+interface IProps {
+    poster: string;
+    streamUrl: string
+}
+
+
+export class HLSVideoComponent extends React.PureComponent<IProps> {
+
+    videoElement: HTMLElement;
+    hls: Hls;
+
+    _initHLS = () => {
```

Comment:
If you want private members you can use `private initHLS() {...}` and the compiler will not allow accessing it outside. If you apply the `key` trick I suggested in another comment, these methods will only be called from one place so you may also inline the code in `componentDidMount` / `componentWillUnmount`